# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/systemservices/PCameras.ipdl
# Commit: ba640de8df47
# Full Hash: ba640de8df47d54906c708eb78f2d000d6f6d7f6
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2024-09-13 21:45:07
# Regressor Bug: 1908245
# File Overlap Count: 1
# Description:
#   Bug 1908245 - Add support for missing requestVideoFrameCallback parameters. r=media-playback-reviewers,webidl,padenot,saschanaz
#   
#   This patch adds support for exposing the capture time, receive time,
#   processing duration and RTP timestamp parameters on the callback for
#   rVFC for WebRTC related elements.
# ==============================================================================

diff -r 2f4f9cce6f7d -r ba640de8df47 dom/media/systemservices/PCameras.ipdl
--- a/dom/media/systemservices/PCameras.ipdl	Fri Sep 13 18:15:18 2024 +0000
+++ b/dom/media/systemservices/PCameras.ipdl	Fri Sep 13 18:19:57 2024 +0000
@@ -7,7 +7,11 @@
 
 include PBackgroundSharedTypes;
 
+include "mozilla/dom/MediaIPCUtils.h";
+
 using mozilla::camera::CaptureEngine from "mozilla/media/CamerasTypes.h";
+using mozilla::TimeStamp from "mozilla/TimeStamp.h";
+using mozilla::media::TimeUnit from "TimeUnits.h";
 
 namespace mozilla {
 namespace camera {
@@ -28,13 +32,15 @@
 // See VideoFrameUtils.h
 struct VideoFrameProperties
 {
+  TimeStamp captureTime;
   // Size of image data within the ShMem,
   // the ShMem is at least this large
   uint32_t bufferSize;
   // From webrtc::VideoFrame
-  uint32_t timeStamp;
+  uint32_t rtpTimeStamp;
   int64_t ntpTimeMs;
   int64_t renderTimeMs;
+  TimeUnit processingDuration;
   // See webrtc/**/rotation.h
   int rotation;
   int yAllocatedSize;